lib/MixedModels/LMM.rb

```diff
@@ -176,12 +194,20 @@ def LMM.from_formula(formula:, data:, weights: nil, offset: 0.0, reml: true,
           "(e.g. (Days || Subj) is equivalent to (1 | Subj) + (0 + Days | Subj))") if formula.include? "||"
     raise(NotImplementedError, "The notation -1 in not supported." +
           "Use 0 instead, in order to denote the exclusion of an intercept term.") if formula.include? "-1"
+    #Other ways you could validate input: Does it contains a "~"?
+    #Does it contain exactly one "~"? Are the rhs and lhs both non-empty?
+    #Are parentheses matched?
+    #Probably you also can't have a variable named intercept?
+
     #remove whitespaces
     formula.gsub!(%r{\s+}, "")
     # replace ":" with "*", because the LMMFormula class requires this convention
     formula.gsub!(":", "*")
     # deal with the intercept (LMM#from_daru handles the case when both, intercept and 
     # no_intercept, are included)
+    #It took me a while to figure out these 4 lines. Maybe add some comments.
+    #1+ is equivalent to intercept+
+    #The 1 is optional when specifying the formula?
```

Comment:
Yes, the intercept term is usually included in linear models by default (having a linear term without an intercept is a statistically flawed practice in most cases), but it needs to be specified manually in pure intercept models like `(1 | g)` or `y ~ 1`.